From 1643cce810c0fb0b2ef9cb370292e4286a70f336 Mon Sep 17 00:00:00 2001
From: Claudia <claui@users.noreply.github.com>
Date: Tue, 19 Sep 2023 17:14:51 +0200
Subject: [PATCH] mmc1: fix duplicate function identifier

Building the `mmc1` example:

> nesfab ./mmc1/mmc1.cfg

results in the following build error:

> ../../../../../../../usr/share/nesfab/lib/mapper/mmc1.fab:94:8: error: Global identifier mmc1_set_chr_1 already in use.
>  94 | asm fn mmc1_set_chr_1(U value)
>              ^^^^^^^^^^^^^^
> ../../../../../../../usr/share/nesfab/lib/mapper/mmc1.fab:69:8: note: Previous definition here:
>  69 | asm fn mmc1_set_chr_1(U value)
>              ^^^^^^^^^^^^^^

Fix the duplicate definition by appending an `_unsafe` suffix, just like
it was done with `mmc1_set_chr_0` vs. `mmc1_set_chr_0_unsafe`.
---
 lib/mapper/mmc1.fab | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/mapper/mmc1.fab b/lib/mapper/mmc1.fab
index 60bd743..8605850 100644
--- a/lib/mapper/mmc1.fab
+++ b/lib/mapper/mmc1.fab
@@ -91,7 +91,7 @@ asm fn mmc1_set_chr_1(U value)
         rts
 
 // Writes the CHR_1 register, the unsafe bank switch version:
-asm fn mmc1_set_chr_1(U value)
+asm fn mmc1_set_chr_1_unsafe(U value)
 : employs
 : +static
         lda &value
